lib/helper.js

```diff
@@ -53,14 +53,19 @@ function parseTemplateConfigName(config) {
 /**
  * recursively fetching records (ex: builds/events) until the end
  * @method getAllRecords
- * @param  {String}       funcName    Function to fetch. For example: 'getBuilds'
- * @param  {Object}       opts        opts to pass into scan function
- * @param  {Array}        resultArray Result so far
- * @param  {String}       date        Current date for aggregation
- * @param  {Number}       index       Current index (each index is a different date)
- * @return {Array}                    Array where each element is an array of records belong to the same date
+ * @param  {String}       funcName            Function to fetch. For example: 'getBuilds'
+ * @param  {String}       aggregateInterval   Aggregate data by day/month/week/year
+ * @param  {Object}       opts                opts to pass into scan function
+ * @param  {Array}        resultArray         Result so far
+ * @param  {String}       date                Current date for aggregation
+ * @param  {Number}       index               Current index (each index is a different date)
+ * @return {Array}                            Array where each element is an array of records belong to the same date
  */
-async function getAllRecords(funcName, opts, resultArray, date, index) {
+async function getAllRecords(funcName, aggregateInterval, opts, resultArray, date, index) {
+    if (!['day', 'week', 'month', 'year'].includes(aggregateInterval)) {
```

Comment:
do we still need to check here? seeing API has this check already for the query param

Reply:
probably don't need. I wrote this before the API, so that's why I had this check. But API already checks so I can remove it 👍